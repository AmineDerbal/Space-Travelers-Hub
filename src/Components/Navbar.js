import { NavLink } from 'react-router-dom';
import './styles/Navbar.css';
import logo from './logo.png';

const Navbar = () => (
  <nav className="nav-bar">
    <div className="logo">
      <img src={logo} className="App-logo" alt="logo" />
      <h2>Space Travelers&apos; Hub</h2>
    </div>
    <ul className="links">
      <li>
        <NavLink to="/">Rockets</NavLink>
      </li>
      <li>
        <NavLink to="/missions">Missions</NavLink>
      </li>
      <span className="separator">|</span>
      <li>
        <NavLink to="/myprofile">My Profile</NavLink>
      </li>
    </ul>
  </nav>
);

export default Navbar;
