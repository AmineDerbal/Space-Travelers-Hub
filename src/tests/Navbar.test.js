import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from '../Components/Navbar';

it('test Navbar rendering', () => {
  render(
    <Router>
      <Navbar />
    </Router>,
  );
  const tree = renderer
    .create(
      <Router>
        <Navbar />
      </Router>,
    )
    .toJSON();
  const title = screen.getByRole('heading', { name: /Space Travelers' Hub/i });
  const navbars = screen.getAllByRole('link');
  expect(title.textContent).toBe("Space Travelers' Hub");
  expect(navbars).toHaveLength(3);
  expect(tree).toMatchSnapshot();
});
