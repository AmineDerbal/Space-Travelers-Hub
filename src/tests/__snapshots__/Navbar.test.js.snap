// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Navbar rendering 1`] = `
<nav
  className="nav-bar"
>
  <div
    className="logo"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.png"
    />
    <h2>
      Space Travelers' Hub
    </h2>
  </div>
  <ul
    className="links"
  >
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        Rockets
      </a>
    </li>
    <li>
      <a
        className=""
        href="/missions"
        onClick={[Function]}
      >
        Missions
      </a>
    </li>
    <span
      className="separator"
    >
      |
    </span>
    <li>
      <a
        className=""
        href="/myprofile"
        onClick={[Function]}
      >
        My Profile
      </a>
    </li>
  </ul>
</nav>
`;
